File: ObjectAvoidance/java/src/main/java/elite/math/trig/Arcsecant.java
package elite.math.trig;

import elite.math.trig.triangle.Adjacent;
import elite.math.trig.triangle.Hypotenuse;

/**
 *
 * @author CFerg (Elite)
 */
public class Arcsecant {

    /**
     *
     */
    public double arcsecant;
    
    /**
     *
     * @param a
     * @param h
     */
    public Arcsecant(Adjacent a, Hypotenuse h){
        this.arcsecant = (a.adjacent / h.hypotenuse);
    }
    
    /**
     *
     * @param sec
     */
    public Arcsecant(Secant sec){
        this.arcsecant = (1 / sec.secant);
    }
    
    /**
     *
     * @param cos
     */
    public Arcsecant(Cosine cos){
        this.arcsecant = cos.cosine;
    }
}

File: ObjectAvoidance/src/elite/math/trig/Arcsecant.kt
package elite.math.trig

import elite.math.trig.triangle.Adjacent
import elite.math.trig.triangle.Hypotenuse

/**
 *
 * @author CFerg (Elite)
 */
class Arcsecant {

    /**
     *
     */
    var arcsecant: Double = 0.toDouble()

    /**
     *
     * @param a
     * @param h
     */
    constructor(a: Adjacent, h: Hypotenuse) {
        this.arcsecant = a.adjacent / h.hypotenuse
    }

    /**
     *
     * @param sec
     */
    constructor(sec: Secant) {
        this.arcsecant = 1 / sec.secant
    }

    /**
     *
     * @param cos
     */
    constructor(cos: Cosine) {
        this.arcsecant = cos.cosine
    }
}